draft-brunstrom-taps-impl.md

```diff
@@ -330,6 +343,35 @@ Protocol options are checked next in order. Whether or not a set of protocol, or
 
 Branching for derived endpoints is the final step, and may have multiple layers of derivation or resolution, such as DNS service resolution and DNS hostname resolution.
 
+
+## Ranking Branches
+
+Implementations SHOULD rank the branches of the tree of connection options in order to prioritize them for candidate racing. Leaf nodes on branches with higher rankings represent connection attempts that will be raced first. Implementations SHOULD order the branches to reflect the preferences expressed by the application for its new connection, including Protocol and Path Selection Properties as well as Application Intents, which are specified in {{draft-trammell-taps-interface}}. In addition to the properties provided by the application, an implementation MAY include additional criteria such as cached performance estimates, see {{performance-caches}}, or system policy, see {{role-of-system-policy}}, in the ranking.
+
+The Transport Parameters specified in {{draft-trammell-taps-interface}} may be used to rank branches in the following ways:
+
+* Interface Type:
+If the application specifies an interface type to be required, an implementation MUST purge all branches that correspond to paths over interfaces that do not match this type. Similarly, it MUST purge paths that correspond to interface types that the application prohibits. If the application specifies an interface type to be preferred or avoided, implementations SHOULD rank paths accordingly.
```

Comment:
I disagree with the approach here: we shouldn't:

1. Gather all possible paths into a tree
2. During sorting, prune out the ones that aren't allowed

Instead, it's simpler to:

1. Gather all permissible paths
2. Sort them for preference

Essentially, the pruning is already covered by Candidate Gathering. We can add more text there to emphasize that (it already has "This list is determined by the requirements, prohibitions, and preferences of the application as specified in the Path Selection Properties and Protocol Selection Properties."), but that's where that operation belongs.

Reply:
Should be addressed now by #8ecb7df